sdk/webpubsub/azure-messaging-webpubsub/src/main/java/com/azure/messaging/webpubsub/WebPubSubAuthenticationPolicy.java

```diff
@@ -64,14 +64,14 @@ AzureKeyCredential getCredential() {
      */
     @Override
     public Mono<HttpResponse> process(final HttpPipelineCallContext context, final HttpPipelineNextPolicy next) {
-        final String audienceUrl = context.getHttpRequest().getUrl().toString();
-        final String token = getAuthenticationToken(audienceUrl, null, credential);
+        return Mono.fromRunnable(() -> {
```

Comment:
Mind leaving a comment on why this is deferred to an asynchronous call even though all operations appear to be synchronous

Reply:
Yes, all operations are synchronous but these operations should be performed at "subscription time" and not during "assembly time".
https://spring.io/blog/2019/03/06/flight-of-the-flux-1-assembly-vs-subscription